mqtt-streaming/src/main/scala/akka/stream/alpakka/mqtt/streaming/impl/RequestState.scala

```diff
@@ -375,22 +376,25 @@ import scala.util.{Failure, Success}
 
   // Processing
 
-  def main(registrantsByPacketId: Map[PacketId, ActorRef[A]], nextPacketId: PacketId): Behavior[Request[A]] =
+  def main(registrantsByPacketId: Map[PacketId, ActorRef[A]], lastPacketId: Option[PacketId]): Behavior[Request[A]] =
     Behaviors.receiveMessage {
-      case Register(registrant: ActorRef[A], reply) if nextPacketId.underlying <= MaxPacketId.underlying =>
+      case Register(registrant: ActorRef[A], reply) if registrantsByPacketId.size < MaxNrOfPacketId =>
+        @tailrec
+        def step(id: PacketId): PacketId =
```

Comment:
Note that if `registrantsByPacketId` ever ends up holding an invalid packet ID (due to bugs introduced by future changes), this will never terminate.